File: src/public/InterfaceStyle.ts
export enum InterfaceStyle {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}

export namespace InterfaceStyle {
  export function fromString(value: string): InterfaceStyle {
    switch (value) {
      case 'LIGHT':
        return InterfaceStyle.LIGHT;
      case 'DARK':
        return InterfaceStyle.DARK;
      default:
        return InterfaceStyle.LIGHT;
    }
  }
}

File: android/src/main/java/com/superwallreactnative/models/InterfaceStyle.kt
package com.superwallreactnative.models

import com.superwall.sdk.network.device.InterfaceStyle

class InterfaceStyle {
  companion object {
    fun fromString(interfaceStyle: String): InterfaceStyle {
      return when (interfaceStyle) {
        "LIGHT" -> InterfaceStyle.LIGHT
        "DARK" -> InterfaceStyle.DARK
        else -> InterfaceStyle.LIGHT // Default case to handle unexpected values
      }
    }
  }
}
